include/xsimd/xsimd.hpp

```diff
@@ -35,6 +35,21 @@
 #define XSIMD_NO_DISCARD
 #endif
 
+#ifdef __cpp_if_constexpr
+// this means that the compiler supports the `if constexpr` construct
+#define XSIMD_IF if constexpr
+#endif
+
+#if !defined(XSIMD_IF) && __cplusplus >= 201703L
```

Comment:
Should be and `#elif  __cplusplus >= 201703L`

Reply:
I was simply following the way we defined `XSIMD_NO_DISCARD`, via a bunch of cascading checks. But if you still want me to change it, I will do so.